File: solutions/javascript/matching-brackets/1/matching-brackets.js
export const isPaired = (text) => {
  const pairs = {
    ')': '(',
    ']': '[',
    '}': '{'
  };
  
  const stack = [];
  
  for (const char of text) {
    // If it's an opening bracket, push to stack
    if (char === '(' || char === '[' || char === '{') {
      stack.push(char);
    }
    // If it's a closing bracket, check for match
    else if (char === ')' || char === ']' || char === '}') {
      // Check if stack is empty or if brackets don't match
      if (stack.length === 0 || stack.pop() !== pairs[char]) {
        return false;
      }
    }
    // Ignore all other characters
  }
  
  // Stack should be empty if all brackets are matched
  return stack.length === 0;
};

File: solutions/python/matching-brackets/1/matching_brackets.py
def is_paired(input_string):
    """
    Check if all brackets, braces, and parentheses are properly matched and nested.
    
    Args:
        input_string: A string containing brackets [], braces {}, parentheses (), 
                     and other characters
    
    Returns:
        True if all pairs are matched and nested correctly, False otherwise
    """
    # Stack to keep track of opening brackets
    stack = []
    
    # Mapping of closing brackets to their corresponding opening brackets
    matching_brackets = {
        ')': '(',
        ']': '[',
        '}': '{'
    }
    
    # Opening brackets
    opening_brackets = set(['(', '[', '{'])
    
    # Process each character in the string
    for char in input_string:
        if char in opening_brackets:
            # Push opening bracket onto stack
            stack.append(char)
        elif char in matching_brackets:
            # It's a closing bracket
            if not stack:
                # No matching opening bracket
                return False
            
            # Check if the top of stack matches
            if stack[-1] == matching_brackets[char]:
                stack.pop()
            else:
                # Mismatched brackets
                return False
        # Ignore all other characters
    
    # If stack is empty, all brackets are matched
    return len(stack) == 0